src/dyn/dyn_variables_electronic.cpp

```diff
@@ -1343,6 +1333,86 @@ vector<double> dyn_variables::compute_average_sh_pop(int rep){
 }
 
 
+
+vector<double> dyn_variables::compute_average_sh_pop_rep_sh0(int rep){

```

Comment:
please add the documentation of the function in the comments block in the beginning - e.g. what' the main purpose of this functions, what's the meaning of the parameter (and possible options), and what it returns.